docs/proposals/kubegen.md

```diff
@@ -0,0 +1,214 @@
+- Extract the tar and add the `kubegen` binary to the PATH
+- Run `kubegen version` to list the version
+  - Will correspond to a Kubernetes release
+
+### Running the command line
+
+#### Use defaults
+
+```sh
+kubegen
+```
+
+- run all code generators against discovered APIs
+  - run for both internal and external types
+- prepend license to generated files
+  - wrap the contents of the LICENSE in comments
+  - exit non-zero if the LICENSE file is missing
+
+#### Specifying copyright owners
+
+The copyright owners for the license can be defined with the `--copyright` flag
+
+```sh
+kubegen --copyright "The Kubernetes Authors."
+```
+
+- run all code generators against discovered APIs
+  - run for both internal and external types
+- prepend license to generated files
+  - wrap the contents of the LICENSE in comments
+  - start license with "Copyright <current year> <copyright>" e.g. *Copyright 2018 The Kubernetes Authors.*
+  - exit non-zero if the LICENSE file is missing
+
+
+#### Specifying the license file
+
+The license file can be overridden from the LICENSE file using the `--license-file` flag
+
+```sh
+kubegen --license-file "boilerplate.go.txt"
+```
+
+- run all code generators against discovered APIs
+  - run for both internal and external types
+- prepend license to generated files
+  - use the contents from boilerplate.go.txt
+  - if boilerplate.go.txt is already in comments then use it verbatim, otherwise wrap it in comments
+  - exit non-zero if boilerplate.go.txt is missing
+
+#### Specifying a license type
+
+The license can be overridden from the LICENSE file using the `--license` flag
+
+```sh
+kubegen --license "Apache 2.0"
+```
+
+- run all code generators against discovered APIs
+  - run for both internal and external types
+- prepend license to generated files
+  - use an *Apache 2.0* license
+  
+#### Generating for external types only
+
+Generating code for internal types can be skipped using the `generate-internal` flag
+
+```sh
+kubegen --generate-internal=false
+```
+
+- run all code generators against discovered APIs
+  - run for external types only
+- prepend license to generated files
+  - wrap the contents of the LICENSE in comments
+  - exit non-zero if the LICENSE file is missing
+
+#### Specifying API versions
+
+Generating code only for specific API group versions can be performed by providing the API
+group versions as positional arguments.
+
+```sh
+kubegen apps/v1 apps/v1beta1 extensions/v1beta1
+```
+
+- run all code generators against the apps/v1 apps/v1beta1 extensions/v1beta1 API groups versions
+  - run for both internal and external types
+- prepend license to generated files
+  - wrap the contents of the LICENSE in comments
+  - exit non-zero if the LICENSE file is missing
+
+#### Specifying generators
+
+Generating code using only a specific set of generators can be performed using the
+`--generator` flag.  This flag may be provided multiple times to run multiple
+generators.
```

Comment:
it's a bit odd that the first `--generator` arg replaces the "all generators" list with a single generator. Then the second appearance appends the generator. IMO a comma separated `--generators` is more natural.

Reply:
This is a common pattern for cobra commands and I believe is preferred.  Comma separated lists are hard when there can be commas in the lists.